src/vector.py

```diff
@@ -4,19 +4,80 @@ class Vector:
   """
   The vector class will create a vector object from a calculated value. It will
   create magnitude, direction, and component attributes.
+
+  Attributes:
+    x : float
+      value of x in rectangular coordinate system
+    y : float
+      value of x in rectangular coordinate system
+    r: float
+      value of radius in polar coordinate system
+    theta : float
+      value of theta (in radians) in polar coordinate system
   """
 
-  def __init__(self, x, y):
+  def __init__(self, *, x=None, y=None, r=None, theta=None):
+    """
+    Initializes a vector from either rectangular or polar coordinate representation.
+    NOTE: Keyword arguments REQUIRED. Either define a vector with x and y keywords for rectangular components, or 
+    r and theta for polar coordinates.  Initializer will throw a ValueError if both coordinate 
+    systems are used.
+    
+    Parameters:
+      x : float
+      value of x in rectangular coordinate system
+      y : float
+        value of x in rectangular coordinate system
+      r: float
+        value of radius in polar coordinate system
+      theta : float
+        value of theta (in radians) in polar coordinate system
     """
-    Initializes a vector from spherical components
-    :param x: x-component
-    :param y: y-component
+
+    if x is not None or y is not None:
+      if r is not None or theta is not None:
+        raise ValueError('Cannot define a Vector by Rectangualr and Polar coordinate systems.  Use x and y or r and theta')
+      else:
+        self.x = x
+        self.y = y
+        self.update_polar(x=self.x, y=self.y)
+    else:
+      if r is not None and theta is not None:
+        self.r = r
+        self.theta = theta
+        self.update_rect(self.r, self.theta)
+      else:
+        raise ValueError('Cannot define a Vector by both Rectangualr and Polar coordinate systems.  Use x and y or r and theta')
+
+
+  def update_polar(self, x: float, y: float) -> None:
```

Comment:
Can we make this a parameterless method and just use self.x and self.y inside the method?  That way we don't run into a possible conflict of state

Reply:
I originally had this parameter-less, not sure why I changed it but had a reason at the time.  I'll change it back until I remember why.